samples/wifi/sta/src/main.c

```diff
@@ -255,6 +285,18 @@ void main(void)
 	k_sleep(K_SECONDS(1));
 
 	wifi_connect();
-	k_sleep(K_SECONDS(30));
+
+	k_thread_start(client_status);
+	k_thread_join(client_status, K_SECONDS(30));
```

Comment:
Why to create a separate thread only to blocking the current one, waiting for the new thread to terminate? Couldn't `cmd_wifi_status()` be printed directly from main?

Reply:
@rlubos Yes, `cmd_wifi_status()` can be printed from main. But, for readbility, we are polling the status in new thread.

Reply:
As the connect is a non-blocking API, we can use main thread itself. Also, the connect starts a status thread anyway, so, if we can just log a couple of debug prints that can be enabled by customer, no need for polling status here, just can add a timeout based on connected.